protocols/872-cosmosid-ngs-library-prep-part2/nexteraXT_dna_library_prep_part2.ot2.py

```diff
@@ -0,0 +1,103 @@
+from opentrons import labware, instruments, modules, robot
+
+"""
+Clean Up Libraries
+"""
+# labware setup
+mag_deck = modules.load('magdeck', '4')
+# mag_plate = labware.load('96-PCR-flat', '4', share=True)
+mag_plate = labware.load('biorad-hardshell-96-PCR', '4', share=True)
+in_plate = labware.load('96-PCR-flat', '5')
+out_plate = labware.load('96-PCR-flat', '1')
+# out_plate = labware.load('biorad-hardshell-96-PCR', '2')
+trough = labware.load('trough-12row', '2')
+liquid_trash = trough.wells('A12')
+
+# reagent setup
+rsb = trough.wells('A1')  # resuspension buffer
+beads = trough.wells('A2')  # AMPure XP beads
+ethanol = trough.wells('A3')  # 80% ethanol
+
+tipracks50 = [labware.load('tiprack-200ul', slot) for slot in ['6', '7', '8']]
+
+# pipette setup
+p50 = instruments.P50_Single(
+    mount='left',
+    tip_racks=tipracks50)
+
+
+def run_custom_protocol(
+        number_of_samples: int=24,
+        PCR_product_volume: float=50,
+        bead_ratio: float=1.8,
+        dry_time: int=15
+        ):
+
+    if number_of_samples <= 24:
+        input = [well
+                 for col in in_plate.cols('1', to='6')
+                 for well in col.wells('A', to='D')][:number_of_samples]
+        mag = [well
+               for col in mag_plate.cols('1', to='6')
+               for well in col.wells('A', to='D')][:number_of_samples]
+        output = [well
+                  for col in out_plate.cols('1', to='6')
+                  for well in col.wells('A', to='D')][:number_of_samples]
+    else:
+        input = [well for well in in_plate.wells()][:number_of_samples]
+        mag = [well for well in mag_plate.wells()][:number_of_samples]
+        output = [well for well in out_plate.wells()][:number_of_samples]
+
+    bead_vol = PCR_product_volume*bead_ratio
+
+    # Transfer PCR product
+    p50.transfer(PCR_product_volume, input, mag, new_tip='always')
+
+    # Transfer beads to each well
+    p50.distribute(bead_vol, beads, [well.top() for well in mag])
+
+    total_vol = bead_vol + PCR_product_volume + 5
+
+    robot.comment("Shake at 1800 rpm for 2 minutes.")
+
+    robot.pause()
```

Comment:
I think this causes weird behavior (at least it used to) when you have a pause right before a delay.

Maybe we want to make all of these delays manual?